board/app/models/thread.php

```diff
@@ -1,122 +1,180 @@
 <?php
 
-class Thread extends AppModel{
-
+class Thread extends AppModel 
+{
 /* 
-*The following constants are declared to avoid magic numbers.
-*Avoid using magic numbers so others couuld
-*   understand what that number is all about.
+* Avoid using magic numbers so others could understand what that number is all about.
 */
-    const MIN_TITLE_LENGTH = 1;
-    const MAX_TITLE_LENGTH = 30;
+const MIN_TITLE_LENGTH = 1;
+const MAX_TITLE_LENGTH = 30;
 
-//Thread Length Validation
+    /* 
+    * Thread Length Validation
+    */
     public $validation = array(
-        'title' => array(
-            'length' => array(
-                'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
+    'title' => array(
+        'length' => array(
+            'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
+            ),
+        ),
+
+    /* 
+    * Category Validation
+    */
+    'category' => array(
+        'length' => array(
+            'validate_between', self::MIN_TITLE_LENGTH, self::MAX_TITLE_LENGTH
             ),
         ),
     );
 
-    public function create(Comment $comment) 
+    public function create(Comment $comment)
     {
         $this->validate();
         $comment->validate();
+        
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('Invalid thread or comment');
+        }
+            
+        try {
+            $db = DB::conn();
+            $date_created = date("Y-m-d H:i:s");
+            $db->begin();
+            $params = array(  
+                'title' => $this->title, 
+                'created'=> $date_created, 
```

Comment:
It would be better to be consistent with naming your variables according to the column_name. In this case, please rename your variable from $date_created to $created:
```
'created' => $created,
```